src/hardware/openbench-logic-sniffer/protocol.c

```diff
@@ -491,6 +477,28 @@ SR_PRIV int ols_receive_data(int fd, int revents, void *cb_data)
 			}
 
 			samples_to_write = devc->rle_count + 1;
+			new_sample_buf_size =
+				4 * MAX(devc->limit_samples,
+					devc->num_samples + samples_to_write);
+
+			if (devc->raw_sample_buf_size < new_sample_buf_size) {
+				unsigned int old_size =
+					devc->raw_sample_buf_size;
+				new_sample_buf_size *= 2;
+				devc->raw_sample_buf =
+					g_try_realloc(devc->raw_sample_buf,
+						      new_sample_buf_size);
+				devc->raw_sample_buf_size = new_sample_buf_size;
+
+				if (!devc->raw_sample_buf) {
+					sr_err("Sample buffer malloc failed.");
+					return FALSE;
+				}
+				/* fill with 1010... for debugging */
+				memset(devc->raw_sample_buf + old_size, 0x82,
+				       new_sample_buf_size - old_size);
```

Comment:
I wonder if we still need that? Also 0x82 is not 1010...

From my point of view, the benefits of having a dynamic buffer and, thus, supporting RLE fully is bigger than the disadvantage of having problems with high-latency network connections. I didn't dive in fully, but the code looks reasonable.

Brainstorming: would it be possible to add a driver-option 'use static buffers' to support high latencies? Not requesting it for this series, maybe it can be added on top. From a glimpse, I'd think this is possible.

Reply:
Ah, indeed, the pattern is wrong. I still consider it useful, though, so I changed it to 0xAA. Especially the most recent version of this file contains changes for a different number of channel groups than 4, which I didn't consider when doing these changes in the first place.

I'm happy that you also find RLE support more worthwhile. Ideally, we would know when to finish reading, but at least 2021 me didn't know how.

About having both buffer options: At this point, I would refrain from adding more options to the code, as this means adding more ways in which things could break. I'd think that looking again into whether there is a way to figure out the buffer size in the first place could be worthwhile instead. I agree that I won't change it as part of this series.